Tests/StatsdClientTests/StatsdClientIPV6Tests.swift

```diff
@@ -0,0 +1,47 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftStatsdClient open source project
+//
+// Copyright (c) 2019 the SwiftStatsdClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftStatsdClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import CoreMetrics
+@testable import StatsdClient
+
+private let port = 9999
+
+class StatsdClientIPV6Tests: XCTestCase {
+
+    func testIPV6Address() throws {
+        let server = TestServer(host: "::1", port: port)
+        XCTAssertNoThrow(try server.connect().wait())
+        defer { XCTAssertNoThrow(try server.shutdown()) }
+        
+        var data = [String]()
+
+        let semaphore = DispatchSemaphore(value: 0)
+        server.onData { _, line in
+            semaphore.signal()
+
+            data.append(line)
```

Comment:
These lines must occur in the opposite order: you have to write to the `data` _first_, and then signal the semaphore.

Reply:
Good point. I initially forgot the `defer`, but I just removed that part altogether. In this test we don't really care about the line that was written, just that the client was able to connect I guess.